data/sqlite_database.py

```diff
@@ -63,13 +63,13 @@ def get_netcdf_files(self, timestamp: List[int], variable: List[str]) -> List[st
             variable = ?
             AND timestamp = ?;
         """
-        for ts in timestamp:
+        for ts in sorted(timestamp):
             for v in variable:
                 self.c.execute(query, (v, ts))
                 file_list.append(self.__flatten_list(self.c.fetchall()))
 
-        # funky way to remove duplicates from the list: https://stackoverflow.com/a/7961390/2231969
-        return list(set(self.__flatten_list(file_list)))
+        # funky way to remove duplicates while preserving order: https://stackoverflow.com/a/39835527
+        return list(dict.fromkeys(self.__flatten_list(file_list))) 
```

Comment:
Nit: Could we write a test to ensure the correct ordering for this in `tests/test_sqlite_database.py`? There might already be a test for that but I can't remember LOL